core/src/main/java/smile/regression/OLS.java

```diff
@@ -376,6 +418,58 @@ public double predict(double[] x) {
 
         return b + Math.dot(x, w);
     }
+    
+    /**
+     * Learn a new instance with online regression.
+     * @param x the training instance. 
+     * @param y the target value.
+     */
+    @Override
+    public void learn(double[] x, double y){
+        System.arraycopy(x, 0, onlineParameters.X.data(), 0, p);
+        onlineParameters.X.set(p, 0, 1);
+        onlineParameters.y = y;
+        updateGamma();
+        updateW();
+    }
+    
+    private void updateGamma(){
+        DenseMatrix x = onlineParameters.X;
+        DenseMatrix gamma = onlineParameters.gamma;
+        
+        double v = 1 + gamma.xax(x.data());
+        if(!Math.isZero(v, .01)){
+            gamma = gamma.sub((gamma.abmm(x.aat().abmm(gamma))).div(v));
+        }else{
+            gamma = gamma.sub((gamma.abmm(x.aat().abmm(gamma))));
+        }
+    }
+    
+    private void updateW(){
+        DenseMatrix x = onlineParameters.X;
+        DenseMatrix w = onlineParameters.W;
+        DenseMatrix gamma = onlineParameters.gamma;
+        double y = onlineParameters.y;
+        
+        double[] delta = gamma.abmm(x).mul(Math.dot(x.data(),w.data())-y).data();
+        for (int i = 0; i < p; i++){
+            this.w[i] -= delta[i];
+            onlineParameters.W.set(i, 0, this.w[i]);
+        }
+        b -= delta[p];
+        onlineParameters.W.set(p, 0, b);
+    }
+    
+    /**
+     * Update the weights with given instance and associated target value.
+     * @param x the training instances.
+     * @param y the target values.
+     */
+    public void learn(double[][] x, double[] y){
```

Comment:
People may be confused with this method and the learning in constructor. Do we have to add this method?